packages/next/src/client/components/redirect.ts

```diff
@@ -3,56 +3,105 @@ import type { ResponseCookies } from '../../server/web/spec-extension/cookies'
 
 const REDIRECT_ERROR_CODE = 'NEXT_REDIRECT'
 
-export enum RedirectType {
-  push = 'push',
-  replace = 'replace',
-}
+export type RedirectType = 'replace' | 'push'
```

Comment:
Not using enums gives better IntelliSense support (Note the gray text is Copilot, not IntelliSense)

Before:
![image](https://github.com/vercel/next.js/assets/18369201/d9d3d4d2-b243-43ed-a639-de018354576b)

After:
![image](https://github.com/vercel/next.js/assets/18369201/9b061d9e-7fc9-4f70-a58f-53678ffa6eb0)